app/Http/Livewire/ManageAccounts.php

```diff
@@ -0,0 +1,171 @@
+            $this->searchQuery ?
+                Individual::whereBlind('name', 'name_index', $this->searchQuery)->get() :
+                Individual::all()
+        );
+        $organizations = new Collection(
+            $this->searchQuery ?
+                Organization::where('name->en', 'like', '%'.$this->searchQuery.'%')
+                    ->orWhere('name->fr', 'like', '%'.$this->searchQuery.'%')->get() :
+                Organization::all()
+        );
+        $regulatedOrganizations = new Collection(
+            $this->searchQuery ?
+                RegulatedOrganization::where('name->en', 'like', '%'.$this->searchQuery.'%')
+                    ->orWhere('name->fr', 'like', '%'.$this->searchQuery.'%')->get() :
+                RegulatedOrganization::all()
+        );
+
+        $accounts = $individuals
+            ->merge($organizations)
+            ->merge($regulatedOrganizations)
+            ->sortBy(fn ($item) => $item->name);
+
+        return view('livewire.manage-accounts', [
+            'accounts' => $accounts->paginate(20),
+        ])
+            ->layout('layouts.app-wide');
+    }
+
+    public function approveIndividualAccount(int $id)
+    {
+        $individual = Individual::find($id);
+        $individual->user->update(['oriented_at' => now()]);
+
+        $individual->user->notify(new AccountApproved($individual));
+
+        $this->dispatchBrowserEvent('clear-flash-message');
+
+        session()->flash('message', __(':account has been approved.', ['account' => $individual->name]));
+
+        $this->dispatchBrowserEvent('add-flash-message');
+
+        $this->dispatchBrowserEvent('remove-flash-message');
+    }
+
+    public function approveAccount(int $id, string $class)
+    {
+        $classname = "App\\Models\\{$class}";
+        $model = $classname::find($id);
+        $model->update(['oriented_at' => now(), 'validated_at' => now()]);
+
+        $model->notify(new AccountApproved($model));
+
+        $this->dispatchBrowserEvent('clear-flash-message');
+
+        session()->flash('message', __(':organization has been approved.', ['organization' => $model->getTranslation('name', locale())]));
+
+        $this->dispatchBrowserEvent('add-flash-message');
+
+        $this->dispatchBrowserEvent('remove-flash-message');
+    }
+
+    public function suspendIndividualAccount(int $id)
+    {
+        $individual = Individual::find($id);
+        $individual->user->update(['suspended_at' => now()]);
+
+        $individual->user->notify(new AccountSuspended($individual));
+
+        $this->dispatchBrowserEvent('clear-flash-message');
+
+        session()->flash('message', __(':account has been suspended.', ['account' => $individual->name]));
+
+        $this->dispatchBrowserEvent('add-flash-message');
+
+        $this->dispatchBrowserEvent('remove-flash-message');
+    }
+
+    public function suspendAccount(int $id, string $class)
+    {
+        $classname = "App\\Models\\{$class}";
```

Comment:
Rather than manually constructing the classname, you could call [`get_class()`](https://www.php.net/manual/en/function.get-class.php). 

Is it possible just to pass in the model instance directly. It looks like you have access to the model when this method is called.

Reply:
I called `get_class()` in the view initially and ran into an issue where the backslashes were being escaped— it's probably easier to do that and handle the escaped values though. I haven't had any luck passing the model instance directly, but I can look into that again.